### PR TITLE
Reworked auto-update and auto-sync with my list

### PR DESCRIPTION
**Pull request by CastagnaIT**
### Check if this PR fulfills these requirements:
* [x] My changes respect the [Kodi add-on development rules](https://kodi.wiki/view/Add-on_rules)
* [x] I have read the [**CONTRIBUTING**](Contributing.md) document.
* [x] I made sure there wasn't another one [Pull Requests](../../pulls) opened for the same update/change
* [x] I have successfully tested my changes locally

#### Types of changes
- [ ] New feature (non-breaking change which adds functionality)
- [x] Feature change (non-breaking change which change behaviour of an existing functionality)
- [x] Improvement (non-breaking change which improve functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### Description

The description is too long, refer to this issue post https://github.com/CastagnaIT/plugin.video.netflix/issues/285
In short:
- No more auto-update linked to auto-sync with my list feature
- No more limitation to use main profile, you can choose which profile to sync with my list
- Auto-update (exports of new seasons/episodes) can be performed manually or by scheduling
- Auto-sync with my list now sync also the movies
- More intuitive settings

Updated Wiki page explains how to use these features
https://github.com/CastagnaIT/plugin.video.netflix/wiki/How-to-export-and-sync-tv-shows-and-movies-in-Kodi-library

#### Describe the new behavior

Users using these features will need to reconfigure their auto-update and sync settings

### CODE REVIEW

**Comment by CastagnaIT** [comment]
@dagwieers
for a few days now i can no longer view the status of Travis CI in all pull requests
but travis compilations are done as usual
can you help me?

**Comment by dagwieers** [comment]
@CastagnaIT Things are being built fine: https://travis-ci.org/CastagnaIT/plugin.video.netflix/builds

**Comment by CastagnaIT** [comment]
Yes via link i can see but
i explain myself better with a screenshot, where there is the arrow is no longer shown
and commits are no longer marked with green circle or X
image
does github no longer provide this "visual" service?

**Comment by dagwieers** [comment]
> does github no longer provide this "visual" service?

It certainly does for other GitHub projects. Probably related to project settings either in GitHub or in Travis CI ? I haven't experienced this before.

**Comment by CastagnaIT** [comment]
can be something related to migrating to travis-ci.com
https://blog.travis-ci.com/2018-05-02-open-source-projects-on-travis-ci-com-with-github-apps

if i go to:
https://travis-ci.com/CastagnaIT/plugin.video.netflix/builds
the build stucks

**Comment by CastagnaIT** [comment]
do not know why but now seem started work again...
image

**Comment by dagwieers** [comment]
Things seem to work again?

**Comment by CastagnaIT** [comment]
yes now returned working as screenshot above ^^
maybe their problems boh

**Comment by notoco** [comment]
Unfortunately - this version (https://github.com/CastagnaIT/plugin.video.netflix/tree/autoupdaterework) on a clean Netflix profile does not start at all. Same on pure kodi
```
plugin.video.netflix (33)] URL is ParseResult(scheme='plugin', netloc='plugin.video.netflix', path='/', params='', query='', fragment='')
2019-11-14 07:53:31.262 T:2936   DEBUG: [plugin.video.netflix (33)] Initialize addon upgrade operations, from version None to 0.15.8)
2019-11-14 07:53:31.263 T:2936   ERROR: [plugin.video.netflix (33)] Traceback (most recent call last):
                                              File "C:\Users\xxxxxxxxxxxxx\AppData\Roaming\Kodi\addons\plugin.video.netflix\addon.py", line 112, in <module>
                                                upgrade_ctrl.check_addon_upgrade()
                                              File "C:\Users\xxxxxxxxxxxxx\AppData\Roaming\Kodi\addons\plugin.video.netflix\resources\lib\upgrade_controller.py", line 18, in check_addon_upgrade
                                                _perform_addon_changes(addon_previous_ver, addon_current_ver)
                                              File "C:\Users\xxxxxxxxxxxxx\AppData\Roaming\Kodi\addons\plugin.video.netflix\resources\lib\upgrade_controller.py", line 40, in _perform_addon_changes
                                                if common.is_less_version(previous_ver, '0.15.9'):
                                              File "C:\Users\xxxxxxxxxxxxx\AppData\Roaming\Kodi\addons\plugin.video.netflix\resources\lib\common\misc_utils.py", line 293, in is_less_version
                                                return list(map(int, version.split('.'))) < list(map(int, max_version.split('.')))
                                            AttributeError: 'NoneType' object has no attribute 'split'
```

https://hastebin.com/lufosojiya.sql

**Comment by CastagnaIT** [comment]
Done thanks!

**Comment by notoco** [comment]
Everything seems to be working. However, it is difficult to do thorough tests, especially in the subject of self-actualization ... I have no idea how to do it .

**Comment by CastagnaIT** [comment]
> Everything seems to be working. However, it is difficult to do thorough tests, especially in the subject of self-actualization ... I have no idea how to do it .

You can explain in a different way? i don't understand what you mean..

**Comment by notoco** [comment]
The automatic library update test would require a TVShow that would have just added new episodes (in Netflix). Then you can check if they are actually added (using autoupdate). If I add the series to the library now, I have all the current episodes, so I can't check for new episodes updates.
How do we say it in Polish: "butter butter" - masło maślane.

**Comment by notoco** [comment]
Auto-sync works :)

**Comment by CastagnaIT** [comment]
haha it sounds strange say "masło maślane" in Italian 😂
for the addition of new episodes, before it worked, so it should work now too
so, that's okay for now
If after the release will be notified problems not seen i will fixit
thank you
